libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -6,398 +6,135 @@ import { ValidationMetadata } from 'class-validator/metadata/ValidationMetadata'
 import { cloneDeep, mergeWith } from 'lodash-es';
 import 'reflect-metadata';
 import { BehaviorSubject } from 'rxjs';
-export interface IShortValidationErrors {
-  [key: string]: string[] | IShortValidationErrors;
-}
+import { Dictionary, ShortValidationErrors, DynamicFormGroupField } from '../models';
+
 export class DynamicFormGroup<TModel> extends FormGroup {
-  customValidateErrors = new BehaviorSubject<IShortValidationErrors>({});
+
+  public customValidateErrors = new BehaviorSubject<ShortValidationErrors>({});
+
   private _object: TModel;
-  private _externalErrors: IShortValidationErrors;
+  private _externalErrors: ShortValidationErrors;
   private _validatorOptions: ValidatorOptions;
   private _fb = new FormBuilder();
-  constructor(
-    public factoryModel: ClassType<TModel>,
-    public fields: {
-      [key: string]: any;
-    },
-    public defaultValidatorOptions?: ValidatorOptions
-  ) {
+
+  constructor(public factoryModel: ClassType<TModel>, public fields: Dictionary, public defaultValidatorOptions?: ValidatorOptions) {
     super({});
-    /*
-    const classValidators = DynamicFormGroup.getClassValidators<TModel>(
-      this.factoryModel,
-      this.fields,
-      this.defaultValidatorOptions
-    );
-    const formGroup = this._fb.group(
-      classValidators
-    );
-    Object.keys(formGroup.controls).forEach(key => {
-      this.addControl(key, formGroup.controls[key]);
-    });
-    this.valueChanges.subscribe(data => {
-      this.validate(
-        undefined,
-        this.defaultValidatorOptions
-      );
-    });*/
     this.fields = this.onlyFields(this.fields);
   }
-  static getClassValidators<TModel>(
```

Comment:
Moved to a function at the end of the file instead of a static method. Allows for functions to be created for encapsulation without needing many static functions which would pollute the API of the class.